backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -142,9 +152,31 @@ private void sendMatchResponse(HttpServletResponse response, Match match) throws
     matchExists.put(JSON_FIRST_PARTICIPANT_USERNAME, match.getFirstParticipantUsername());
     matchExists.put(JSON_SECOND_PARTICIPANT_USERNAME, match.getSecondParticipantUsername());
     matchExists.put(JSON_DURATION, match.getDuration());
-
+    try {
+      emailNotifier = createOrGetEmailNotifier();
+    } catch (GeneralSecurityException e) {
+      e.printStackTrace();
+    }
+    try {
+      emailNotifier.sendMatchEmail(getUsername(), userService.getCurrentUser().getEmail());
+    } catch (MessagingException e) {
+      e.printStackTrace();
+    }
     // Send the JSON back as the response
     response.setContentType("application/json");
     response.getWriter().println(matchExists.toString());
   }
+
+  /** Retrieve user email address via Users API and parse for username */
+  private String getUsername() {
```

Comment:
Once thing we've started to use in other parts of the code are nullability annotations. Adding the @Nullable annotation to this method lets callers know that this method can return null. Even better, it hints to static analysis tools like linters that the return value could be null so they'll warn you if you try to dereference getUsername() without a null check. Please add `@Nullable`

Reply:
Ack